engine/xml-config/src/main/java/org/datacleaner/job/JaxbJobReader.java

```diff
@@ -533,13 +534,22 @@ private void wireOutputDataStreams(final Map<ComponentType, ComponentBuilder> co
 
                 for (int i = 0; i < sourceColumnTypes.size(); i++) {
                     final ColumnType sourceColumnPath = sourceColumnTypes.get(i);
-                    final String outputStreamColumnPathName =
-                            getOutputStreamColumnPath(sourceColumnTypes.get(i).getPath(), componentType,
-                                    componentBuilder, i);
+                    final List<MetaModelInputColumn> findSourceColumn = sourceColumns.stream().filter(
+                            sourceCol -> sourceCol.getName().equals(sourceColumnPath.getPath())).collect(Collectors
+                                    .toList());
+                    final String outputStreamColumnPathName;
+                    if (findSourceColumn.size() > 0) {
+                        outputStreamColumnPathName = getOutputStreamColumnPath(findSourceColumn.get(0).getName(),
+                                componentType, componentBuilder, i);
+                    } else {
+                        outputStreamColumnPathName = getOutputStreamColumnPath(sourceColumns.get(i).getName(),
+                                componentType, componentBuilder, i);
```

Comment:
I think I've actually seen it happening: I was using the job we played with yesterday (country standardizer->completeness->csv for 3 columns) to go forward and back from orderdb to customers.csv a few times, and all of a sudden columns were missing.

There _is_ a small chance that it was human error, though. When using a CSV file, _all_ columns will be added to the job, so they need to be removed again to make the mapping easier. I don't think so, though, since they're still in the mapping.